examples/server.rs

```diff
@@ -1,21 +1,66 @@
-use hyper::rt::{self, Future};
-use hyper::service::service_fn_ok;
-use hyper::{Body, Request, Response, Server};
+use futures::{future, Future, Poll, Stream};
+use hyper::{Body, Request, Response};
+use tokio_tcp::TcpListener;
+use tower_hyper::server::Server;
+use tower_service::Service;
 
 fn main() {
     pretty_env_logger::init();
-    let addr = ([127, 0, 0, 1], 3000).into();
 
-    let server = Server::bind(&addr)
-        .serve(|| {
-            // This is the `Service` that will handle the connection.
-            // `service_fn_ok` is a helper to convert a function that
-            // returns a Response into a `Service`.
-            service_fn_ok(move |_: Request<Body>| Response::new(Body::from("Hello World!")))
-        })
-        .map_err(|e| eprintln!("server error: {}", e));
+    hyper::rt::run(future::lazy(|| {
```

Comment:
small nit—can the future inside of of `future::lazy(|| {})` be assigned to a value and then passed into `hyper::rt::run`?

Reply:
done! thanks